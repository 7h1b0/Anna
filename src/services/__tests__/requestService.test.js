const requestService = require('../requestService');

describe('requestService', () => {
  describe('isSuccess', () => {
    it('should return true when status is >= 200 and < 300', () => {
      expect(requestService.isSuccess(200)).toBeTruthy();
      expect(requestService.isSuccess(201)).toBeTruthy();
      expect(requestService.isSuccess(204)).toBeTruthy();
    });

    it('should return false is status > 300', () => {
      expect(requestService.isSuccess(301)).toBeFalsy();
      expect(requestService.isSuccess(500)).toBeFalsy();
      expect(requestService.isSuccess(404)).toBeFalsy();
    });
  });

  describe('getRoute', () => {
    it('should return an object #1', () => {
      expect(
        requestService.getRoute('http://anna.com', 'GET'),
      ).toMatchSnapshot();
    });

    it('should return an object #2', () => {
      expect(
        requestService.getRoute('http://anna.com/test', 'post', { on: true }),
      ).toMatchSnapshot();
    });
  });
});
