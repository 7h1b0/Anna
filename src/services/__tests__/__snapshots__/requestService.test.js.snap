// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestService getRoute should return an object #1 1`] = `
Object {
  "headers": Object {
    "Accept": "application/json",
  },
  "hostname": "anna.com",
  "method": "GET",
  "path": "/",
  "port": null,
  "protocol": "http:",
  "timeout": 2000,
}
`;

exports[`requestService getRoute should return an object #2 1`] = `
Object {
  "body": Object {
    "on": true,
  },
  "headers": Object {
    "Accept": "application/json",
  },
  "hostname": "anna.com",
  "method": "POST",
  "path": "/test",
  "port": null,
  "protocol": "http:",
  "timeout": 2000,
}
`;
